src/FlightDisplay/VideoManager.cc

```diff
@@ -77,6 +77,7 @@ VideoManager::setToolbox(QGCToolbox *toolbox)
     _updateSettings();
     if(isGStreamer()) {
         _videoReceiver->start();
+        _subtitleWriter = new SubtitleWriter();
```

Comment:
You are manually allocating memory. You must also deallocate it.

```suggestion
        if (_subtitleWriter) {
            delete _subtitleWriter;
        }
        _subtitleWriter = new SubtitleWriter;
```

Also in the constructor:
```
delete _subtitleWriter
```

HOWEVER, this class member `_subtitleWriter` is static (it does not need to be changed/reinitialized at any point during the VideoReceiver owner's lifetime). Therefore, `_subtitleWriter` should be declared as:
```
SubtitleWriter _subtitleWriter; // non-pointer type
```